modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/sql/IgniteAbstractSqlAlterTable.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.internal.processors.query.calcite.sql;
+
+import java.util.Objects;
+import org.apache.calcite.sql.SqlDdl;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlSpecialOperator;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+
+/**
+ * Parse tree for {@code ALTER TABLE } statement
+ */
+public abstract class IgniteAbstractSqlAlterTable extends SqlDdl {
```

Comment:
I wonder if you've considered an option to extend SqlAlter class?

Reply:
`SqlAlter` class not extends `SqlDdl`, and is used only for `ALTER (SYSTEM|SESSION)` statements. It has only unnecessary for `ALTER TABLE` additional `scope` field compared to `SqlDdl`, but to use `SqlAlter` we should modify `DdlSqlToCommandConverter` and underlying usages to accept `SqlCall` instead of `SqlDdl`, I'm not sure is it worth it.